### PR TITLE
Add CommandLineParsing framework

### PR DESCRIPTION
**Pull request by awoods**
Resolves: https://jira.duraspace.org/browse/FCREPO-2138

### CODE REVIEW

**Comment by ajs6f** [comment]
This isn't bad, but consider using [Airline](https://github.com/airlift/airline) for more concision and clarity.

File: README.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ Additional requirements for building and running from source:
 To run the standalone utility:
 
 ```sh
-$ java -jar fcrepo-import-export-<version>.jar
+$ java -jar fcrepo-import-export-<version>.jar --mode [import|export]
```
Comment (bseeger) [inline README.md]: Does java automatically convert `--mode` to the `-m` you're looking for below?

Comment (bseeger) [inline README.md]: Never mind, I see how it works.

**Comment by awoods** [comment]
Re: Airline, I will not have the time right now to investigate redoing the CLI support... but if someone else is interested, great.
In a two week sprint, time is of the essence. I would suggest moving forward with this apache/commons-cli, and potentially revisit later.

**Comment by escowles** [comment]
I agree with @awoods here — this looks perfectly functional, and I'd rather get something merged so we can move forward, and reconsider whether something like Airline would be better at a later date.

**Comment by ajs6f** [comment]
I'll go so far as to file a ticket, but I'm certainly not going to do any work for it.

**Pull request merged**